### PR TITLE
Cleanup metadata

### PR DESCRIPTION
**Pull request by hisaac**
### Checklist

- [x] I've read and followed the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- [x] `step.yml` and `README.md` is updated with the changes (if needed)

### Version

~~Requires a *MAJOR/MINOR/PATCH* [version update](https://semver.org/)~~

Since this has not been released to the steplib yet, I will remove the 0.1.0 tag, and target it to the commit generated by this PR.

### Context

This change comes from comments by @lpusok and @koral-- in my PR to add this Step to the steplib. https://github.com/bitrise-io/bitrise-steplib/pull/3522

### Changes

- Adds the `audit-this-step` and `share-this-step` workflows back to the `bitrise.yml` for later use.
- Removes deprecated and unnecessary properties from the `step.yml`.

### CODE REVIEW

File: bitrise.yml [hunk 1]
```diff
@@ -63,3 +63,54 @@ workflows:
             rm -rf ./_tmp
             rm -rf "$BITRISE_DEPLOY_DIR/app-debug.apk"
             rm -rf "$BITRISE_DEPLOY_DIR/app-debug-androidTest.apk"
+
+  # ----------------------------------------------------------------
+  # --- workflows to Share this step into a Step Library
+  audit-this-step:
```
Comment (godrei) [inline bitrise.yml]: This is included in the `check` workflow: https://github.com/bitrise-steplib/steps-check/blob/master/checks.bitrise.yml#L18.
Do you think we still need this WF?

File: bitrise.yml [hunk 2]
```diff
@@ -63,3 +63,54 @@ workflows:
...
+    steps:
+    - script:
+        inputs:
+        - content: |-
+            #!/bin/bash
+            set -ex
+            stepman audit --step-yml ./step.yml
+
+  share-this-step:
```
Comment (godrei) [inline bitrise.yml]: This is only needed for the first version share, later on, unified-ci automates new version releases.
Do you think we still need this WF?

File: bitrise.yml (hunk 1 above)
Comment (hisaac) [reply bitrise.yml]: Ah that makes sense. I can remove it.

File: bitrise.yml (hunk 2 above)
Comment (hisaac) [reply bitrise.yml]: Good point that future versions will be automated. I can remove it.

File: bitrise.yml (hunk 1 above)
Comment (hisaac) [reply bitrise.yml]: Reverted in https://github.com/bitrise-steplib/bitrise-step-android-instrumented-test/pull/7/commits/ba314aafcaacbaaf5aea281fdd478034cc518088

File: bitrise.yml (hunk 2 above)
Comment (hisaac) [reply bitrise.yml]: Reverted in https://github.com/bitrise-steplib/bitrise-step-android-instrumented-test/pull/7/commits/ba314aafcaacbaaf5aea281fdd478034cc518088

**Review by godrei — APPROVED** [review]

**Pull request merged**